### PR TITLE
DEV: Various improvements and refactoring

### PR DESCRIPTION
**Pull request by OsamaSayegh**
This PR makes a few improvements to this component such as:

1. Remove patches to the `topic-list-item` and `latest-topic-list-item` core components
2. Reduce duplicate code
3. Add various linting/code format config files and lint/format the files

Core PR https://github.com/discourse/discourse/pull/23431 is required before this PR can be merged.

### CODE REVIEW

File: javascripts/discourse/connectors/above-topic-list-item/clickable-topic-row.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const topic = this.args.outletArgs.topic;
+
+    const clickTargets = [
+      "topic-list-data",
+      "link-bottom-line",
+      "topic-list-item",
+    ];
+
+    if (this.site.mobileView) {
+      clickTargets.push("right", "clearfix");
```
Comment (nbianca) [inline javascripts/discourse/connectors/above-topic-list-item/clickable-topic-row.js:23]: I find these two classes a bit too generic for targeting and I am wondering if we should add in core a more descriptive class to the element we want to target for the mobile view, like we have for desktop.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "TODO",
+  "version": "0.0.1",
+  "repository": "TODO",
```
Comment (nbianca) [inline package.json:4]:
```suggestion
  "repository": "https://github.com/discourse/discourse-clickable-topic",
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "TODO",
```
Comment (nbianca) [inline package.json:2]:
```suggestion
  "name": "discourse-clickable-topic",
```

File: README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Discourse Clicable Topic Component
 This component makes the entire area of a `topic-list-item` clickable, as opposed to just the title.
 
-![Discourse Clickable Topic Preivew](readme-assets/preview.gif)
+![Discourse Clickable Topic Preivew](https://github.com/discourse/discourse-clickable-topic/assets/17474474/720acb71-b283-4a3a-ad98-39a0d502ca1e)
```
Comment (nbianca) [inline README.md:4]: Why this change? I think that GitHub assets URL works just because we used to have that preview.gif in this repository. I would keep it the way it was.

File: README.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
 
 [Preview on theme-creator](https://theme-creator.discourse.org/theme/jordan.vidrine/clickable-topic 
-)
\ No newline at end of file
+)
```
Comment (nbianca) [inline README.md:7]: Why is this on a new line? This should be at the end of the previous one, right?

**Review by nbianca — APPROVED** [review]
Just a few minor comments.

File: README.md (hunk 4 above)
Comment (OsamaSayegh) [reply README.md:4]: I got the assets URL by uploading the GIF to a post in a PR for this repo and then copying the URL of the image. It's a known trick for hosting/embedding images in a README without cluttering the repository itself with image files that are only needed for the README. We also get to reduce the theme size by about 1.4MB, which is not a huge amount of data, but it's a nice win for no downsides.
